doc/releases/changelog-dev.md

```diff
@@ -713,6 +713,9 @@
 
 <h3>Bug fixes 🐛</h3>
 
+* Fixes `qml.operation.gen_is_multi_term_hamiltonian` to work with more complicated generators.
```

Comment:
@albi3ro do you mind adding a first sentence that describes the main user-facing bug that is now resolved? E.g., 
```suggestion

* Fixes a bug that caused the output of `qml.fourier.qnode_spectrum()` to
  differ depending if equivalent gate generators are defined using
  different PennyLane operators. This was resolved by updating
  `qml.operation.gen_is_multi_term_hamiltonian` to work with more complicated generators.
```